packages/authentication/src/__tests__/fixtures/strategies/jwt-strategy.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {inject} from '@loopback/context';
+import {HttpErrors, Request} from '@loopback/rest';
+import {AuthenticationStrategy, UserProfile} from '../../../types';
+import {JWTAuthenticationStrategyBindings} from '../keys';
+import {JWTService} from '../services/jwt-service';
+
+export class JWTAuthenticationStrategy implements AuthenticationStrategy {
+  name: string = 'jwt';
+
+  constructor(
+    @inject(JWTAuthenticationStrategyBindings.TOKEN_SERVICE)
+    public token_service: JWTService,
+  ) {}
+
+  async authenticate(request: Request): Promise<UserProfile | undefined> {
+    const token: string = this.extractCredentals(request);
+    const userProfile: UserProfile = await this.token_service.verifyToken(
+      token,
+    );
+    return userProfile;
+  }
+
+  extractCredentals(request: Request): string {
+    if (!request.headers.authorization) {
+      //throw an error
+      throw new HttpErrors.Unauthorized(`Authorization header not found.`);
+    } //if
+
+    // for example : Bearer xxx.yyy.zzz
+    let auth_header_value = request.headers.authorization;
+
+    if (!auth_header_value.startsWith('Bearer')) {
+      //throw an error
+      throw new HttpErrors.Unauthorized(
+        `Authorization header is not of type 'Bearer'.`,
+      );
+    } //if
+
+    //split the string into 2 parts : 'Bearer ' and the `xxx.yyy.zzz`
+    let parts = auth_header_value.split(' ');
```

Comment:
Do we want to assert that length of parts does not exceed 2 here?

Reply:
I will add a check, but we must remember these are mock implementations in src/__tests__ and not production code. We don't need to test each line of code for these items. This PR has mocha tests that focus on verifying the authentication strategy interface , registering and finding of authentication strategies via extensionPoints, the authentication action, and the authentication strategy provider. The mock implementations for token service and user service and basic strategy and jwt strategy only serve as fixture helpers and don't need to be tested so heavily.